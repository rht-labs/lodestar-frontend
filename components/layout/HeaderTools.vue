<template>
  <div class="pf-c-page__header-tools">
    <div class="pf-c-page__header-tools-group pf-m-icons">
      <button class="pf-c-button pf-m-plain" type="button" aria-label="Alerts">
        <i class="fas fa-bell" aria-hidden="true"></i>
      </button>
      <button
        class="pf-c-button pf-m-plain"
        type="button"
        aria-label="Settings"
      >
        <i class="fas fa-cog" aria-hidden="true"></i>
      </button>
    </div>
    <div class="pf-c-page__header-tools-group">
      <div class="pf-c-dropdown">
        <button
          id="page-default-nav-dropdown"
          class="pf-c-dropdown__toggle pf-m-mobile pf-m-plain"
          aria-expanded="false"
          aria-label="Actions"
        >
          <i class="fas fa-ellipsis-v" aria-hidden="true"></i>
        </button>
        <ul
          class="pf-c-dropdown__menu pf-m-align-right"
          aria-labelledby="page-default-nav-dropdown"
          hidden
        >
          <li>
            <a class="pf-c-dropdown__menu-item" href="#">Link</a>
          </li>
          <li>
            <button class="pf-c-dropdown__menu-item">Action</button>
          </li>
          <li>
            <a
              class="pf-c-dropdown__menu-item pf-m-disabled"
              aria-disabled="true"
              tabindex="-1"
              href="#"
            >
              Disabled link
            </a>
          </li>
          <li>
            <button class="pf-c-dropdown__menu-item" disabled>
              Disabled action
            </button>
          </li>
          <li class="pf-c-divider" role="separator"></li>
          <li>
            <a class="pf-c-dropdown__menu-item" href="#">Separated link</a>
          </li>
        </ul>
      </div>
      <div class="pf-m-user">
        <div class="pf-c-dropdown">
          <button
            id="page-default-nav-example-dropdown-button"
            ref="userButton"
            class="pf-c-dropdown__toggle pf-m-plain"
            aria-expanded="false"
            @click="toggleDropdown"
          >
            <span class="pf-c-dropdown__toggle-text">{{ name }}</span>
            <i
              class="fas fa-caret-down pf-c-dropdown__toggle-icon"
              aria-hidden="true"
            ></i>
          </button>
          <div
            v-if="showDropdown"
            v-closable="{
              exclude: ['userButton'],
              handler: 'onUnfocus'
            }"
            class="pf-c-dropdown__menu"
          >
            <ul>
              <li @click="logout">
                <p class="pf-c-dropdown__menu-item">Log Out</p>
              </li>
              <!--
              <li>
                <button class="pf-c-dropdown__menu-item">Action</button>
              </li>
              <li>
                <a
                  class="pf-c-dropdown__menu-item pf-m-disabled"
                  aria-disabled="true"
                  tabindex="-1"
                  href="#"
                >
                  Disabled link
                </a>
              </li>
              <li>
                <button class="pf-c-dropdown__menu-item" disabled>
                  Disabled action
                </button>
              </li>
              <li class="pf-c-divider" role="separator"></li>
              <li>
                <a class="pf-c-dropdown__menu-item" href="#">Separated link</a>
              </li>
              -->
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  data: () => {
    return {
      showDropdown: false
    }
  },
  computed: {
    name() {
      return this.$store.state.auth.user.name
    }
  },
  methods: {
    toggleDropdown() {
      this.showDropdown = !this.showDropdown
    },
    onUnfocus() {
      this.showDropdown = false
    },
    logout() {
      this.$auth.logout()
      this.$router.push({
        path: '/login'
      })
    }
  }
}
</script>
