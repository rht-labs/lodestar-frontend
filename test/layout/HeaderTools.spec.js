import { createLocalVue, shallowMount } from '@vue/test-utils'
import Vuex from 'vuex'
import HeaderTools from '@/components/layout/HeaderTools.vue'

const localVue = createLocalVue()
localVue.use(Vuex)
let mockStore
let wrapper

describe('HeaderTools', () => {
  beforeEach(() => {
    mockStore = new Vuex.Store({
      modules: {
        auth: {
          namespaced: false,
          state: { user: { name: 'name' } }
        }
      }
    })
  })

  test('is a Vue instance', () => {
    wrapper = shallowMount(HeaderTools, { store: mockStore, localVue })
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('renders properly', () => {
    wrapper = shallowMount(HeaderTools, { store: mockStore, localVue })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
