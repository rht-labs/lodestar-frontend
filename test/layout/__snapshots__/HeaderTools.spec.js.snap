// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderTools renders properly 1`] = `
"<div class=\\"pf-c-page__header-tools\\"><div class=\\"pf-c-page__header-tools-group pf-m-icons\\"><button type=\\"button\\" aria-label=\\"Alerts\\" class=\\"pf-c-button pf-m-plain\\"><i aria-hidden=\\"true\\" class=\\"fas fa-bell\\"></i></button> <button type=\\"button\\" aria-label=\\"Settings\\" class=\\"pf-c-button pf-m-plain\\"><i aria-hidden=\\"true\\" class=\\"fas fa-cog\\"></i></button></div> <div class=\\"pf-c-page__header-tools-group\\"><div class=\\"pf-c-dropdown\\"><button id=\\"page-default-nav-dropdown\\" aria-expanded=\\"false\\" aria-label=\\"Actions\\" class=\\"pf-c-dropdown__toggle pf-m-mobile pf-m-plain\\"><i aria-hidden=\\"true\\" class=\\"fas fa-ellipsis-v\\"></i></button> <ul aria-labelledby=\\"page-default-nav-dropdown\\" hidden=\\"hidden\\" class=\\"pf-c-dropdown__menu pf-m-align-right\\"><li><a href=\\"#\\" class=\\"pf-c-dropdown__menu-item\\">Link</a></li> <li><button class=\\"pf-c-dropdown__menu-item\\">Action</button></li> <li><a aria-disabled=\\"true\\" tabindex=\\"-1\\" href=\\"#\\" class=\\"pf-c-dropdown__menu-item pf-m-disabled\\">
            Disabled link
          </a></li> <li><button disabled=\\"disabled\\" class=\\"pf-c-dropdown__menu-item\\">
            Disabled action
          </button></li> <li role=\\"separator\\" class=\\"pf-c-divider\\"></li> <li><a href=\\"#\\" class=\\"pf-c-dropdown__menu-item\\">Separated link</a></li></ul></div> <div class=\\"pf-m-user\\"><div class=\\"pf-c-dropdown\\"><button id=\\"page-default-nav-example-dropdown-button\\" aria-expanded=\\"false\\" class=\\"pf-c-dropdown__toggle pf-m-plain\\"><span class=\\"pf-c-dropdown__toggle-text\\">name</span> <i aria-hidden=\\"true\\" class=\\"fas fa-caret-down pf-c-dropdown__toggle-icon\\"></i></button> <!----></div></div></div></div>"
`;
